drivers/filesystems/btrfs/crc32c-gas.S

```diff
@@ -17,8 +17,105 @@
 
 #include <asm.inc>
 
-EXTERN _crctable:DWORD
```

Comment:
Question for these ASM files: Looking at upstream winbtrfs there are separate MASM and GAS files. Here you've kept our synthetic ASM file that can be compiled at the same time on both MASM and GAS...
So what's the best plan here? Keep that file, or switch back to the separate files?

Reply:
here we have 2 possibilities:
 * match upstream and have separate files for MASM and GAS.
 * continue to use our "synthetic" one.
I currently used the second one, but I have to admit the first one is
1) easier to maintain
2) closer to upstream

I'll change it to 2 separate files

Reply:
I changed the name of the *.S to remove the "gas" from the name to avoid confusion.
compilation for MSVC has been fixed